articles/dms/howto-mysql-to-azure-mysql-powershell.md

```diff
@@ -0,0 +1,657 @@
+[securestring] $SourcePassword = Read-Host "Enter MySQL Source Server Password" -AsSecureString
+
+[string] $TargetServerName = "migdocdevwus2mysqlsstrgt.mysql.database.azure.com"
+[string] $TargetUserName = "docadmin@migdocdevwus2mysqlsstrgt"
+[securestring] $TargetPassword = Read-Host "Enter MySQL Target Server Password" -AsSecureString
+
+function InitConnection(
+    [string] $ServerName,
+    [string] $UserName,
+    [securestring] $Password) {
+    $connectionInfo = @{
+        "dataSource"             = "";
+        "serverName"             = "";
+        "port"                   = 3306;
+        "userName"               = "";
+        "password"               = "";
+        "authentication"         = "SqlAuthentication";
+        "encryptConnection"      = $true;
+        "trustServerCertificate" = $true;
+        "additionalSettings"     = "";
+        "type"                   = "MySqlConnectionInfo" 
+    }
+
+    $connectionInfo.dataSource = $ServerName;
+    $connectionInfo.serverName = $ServerName;
+    $connectionInfo.userName = $UserName;
+    $connectionInfo.password = (ConvertFrom-SecureString -AsPlainText $password).ToString();
+    $connectionInfo;
+}
+
+# Initialize the source and target connections
+LogMessage -Message "Initializing source and target connection objects ..." -IsProcessing $true
+$sourceConnInfo = InitConnection `
+    $SourceServerName `
+    $SourceUserName `
+    $SourcePassword;
+
+$targetConnInfo = InitConnection `
+    $TargetServerName `
+    $TargetUserName `
+    $TargetPassword;
+
+LogMessage -Message "Source and target connection object initialization complete."
+```
+
+## Step 7: Extract the list of table names from the target database
+
+Database table list can be extracted using a migration task and connection information. The table list will be extracted from both the source database and target database so that proper mapping and validation can be done. 
+
+The following script takes the names of the source and target databases and then extracts the table list from the databases using the *GetUserTablesMySql* migration task. 
+
+```powershell
+# Run scenario to get the tables from the target database to build
+# the migration table mapping
+[string] $TargetDatabaseName = "migtargetdb"
+[string] $SourceDatabaseName = "migsourcedb"
+
+function RunScenario([object] $MigrationService, 
+    [object] $MigrationProject, 
+    [string] $ScenarioTaskName, 
+    [object] $TaskProperties, 
+    [bool] $WaitForScenario = $true) {
+    # Check if the migration task already exists, if so remove it
+    LogMessage -Message "Removing scenario if already exists..." -IsProcessing $true
+    Remove-AzDataMigrationTask `
+        -ResourceGroupName $MigrationService.ResourceGroupName `
+        -ServiceName $MigrationService.Name `
+        -ProjectName $MigrationProject.Name `
+        -TaskName $ScenarioTaskName `
+        -Force;
+
+    # Start the new migration using the provided properties
```

Comment:
This comment isn't accurate, since the function is used to run any scenario, not just migration scenarios.

Reply:
Done

Reply:
Done